jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/Jackson2Annotator.java

```diff
@@ -129,9 +136,13 @@ public void objectBuilder(JDefinedClass clazz, JDefinedClass builder) {
                         constructor.param(field.type(), field.name())
                                 .annotate(annotation.getAnnotationClass())
                                 .param("value", str);
-                        body.assign(JExpr._this().ref(field.name()),
-                                // TODO(micah): this should respect immutability config
-                                immutableCopy(assignment, field.type()));
+
+                        if (this.getGenerationConfig().isImmutable()) {
+                            body.assign(JExpr._this().ref(field.name()),
+                                    immutableCopy(assignment, field.type()));
+                        } else {
+                            body.assign(JExpr._this().ref(field.name()), assignment);
```

Comment:
I've commented on one of those other cases (in ObjectRule) which I think would also benefit from better separating out the differences between the if/else treatments. Top my eye, PropertyRule and AdditionalPropertiesRule are ok, b/c they're both doing single-branched ifs. (As an aside, it really looks like there's duplicated logic between AdditionalPropertiesRule and PropertyRule, when constructing the modifiers for a field.)